### ISSUE TITLE
Travis is failing on OSX

### ISSUE THREAD

**Issue by wbhart** [wbhart/flint2#321]
@isuruf do you have any idea what is causing this failure?

https://api.travis-ci.org/jobs/202507368/log.txt?deansi=true

**Comment by isuruf** [comment]
Yes, I've seen it before. From OS X 10.11, `DYLD_LIBRARY_PATH` is disabled due to security concerns. The tests created in flint2 don't use rpath and rely on `DYLD_LIBRARY_PATH` to find `libflint.dylib`. One option would be to use rpath when linking the tests or use OS X < 10.11

**Comment by wbhart** [comment]
Ah, thanks for the explanation.

**Comment by isuruf** [comment]
Adding `export LDFLAGS="-Wl,-rpath,/path/to/flint2/root ${LDFLAGS}"` before running make check should fix this. I'm not sure how to fix the build system though

**Comment by wbhart** [comment]
Yes, we should do that. Do you mean you don't know how to make this change
to the build system. Or is there another issue you are referring to?

Probably the above change can be made in Makefile.in, in the check target.
I think Makefile.subdirs is invoked directly using make -C. Exporting
LDFLAGS before that make -C invocation in Makefile.in should be enough to
sort out Makefile.subdirs as well.

**Comment by isuruf** [comment]
Yes, I meant I don't know how to make this change in the build system. Also, I don't know whether it works on all compilers supported by flint.

**Comment by wbhart** [comment]
I think Flint only supports gcc and clang. MSVC is supported via solution
files isn't it?

**Comment by isuruf** [comment]
Right. Then we can use the rpath solution
